MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult+Internal.swift

```diff
@@ -26,20 +26,56 @@ import Foundation
-            return await credentialsController.refreshToken(context: context, authTokens: authTokens, userAccountResult: self)
-        } else {
-            return .init(.success(MSALNativeAuthTokenResult(authTokens: authTokens)), correlationId: correlationId)
+        params.forceRefresh = forceRefresh
+        params.correlationId = correlationId
+
+        let challengeTypes = MSALNativeAuthPublicClientApplication.convertChallengeTypes(configuration.challengeTypes)
+        let authority = try? MSALCIAMAuthority(url: configuration.authority.url)
+        let config = MSALPublicClientApplicationConfig(clientId: configuration.clientId,
+                                                       redirectUri: configuration.redirectUri,
+                                                       authority: authority)
+
+        guard let client = try? MSALNativeAuthPublicClientApplication(configuration: config, challengeTypes: challengeTypes)
+        else {
+            MSALLogger.log(
+                            level: .error,
+                            context: context,
+                            format: "Config or challenge types unexpectedly found nil."
+                        )
+            Task { await delegate.onAccessTokenRetrieveError(
+                error: RetrieveAccessTokenError(type: .generalError,
+                                                correlationId: correlationId ?? context.correlationId())) }
+            return
+        }
+
+        client.acquireTokenSilent(with: params) { result, error in
+
+            if let error = error as? NSError {
+                let accessTokenError = RetrieveAccessTokenError(type: .generalError,
+                                                                correlationId: result?.correlationId ?? UUID(),
+                                                                errorCodes: [error.code])
+                Task { await delegate.onAccessTokenRetrieveError(error: accessTokenError) }
+                return
+            }
+
+            if let result = result {
+                let delegateDispatcher = CredentialsDelegateDispatcher(delegate: delegate, telemetryUpdate: nil)
+                let accessTokenResult = MSALNativeAuthTokenResult(accessToken: result.accessToken,
+                                                                  scopes: result.scopes,
+                                                                  expiresOn: result.expiresOn)
+                Task { await delegateDispatcher.dispatchAccessTokenRetrieveCompleted(result: accessTokenResult, correlationId: result.correlationId) }
+                return
+            }
+
+            Task { await delegate.onAccessTokenRetrieveError(error: RetrieveAccessTokenError(type: .generalError,
+                                                                                             correlationId: correlationId ?? UUID())) }
```

Comment:
can you use the context here?
```suggestion
                                                                                             correlationId: correlationId ?? context.correlationId())) }
```

Reply:
Sure, done.